### PR TITLE
dialects: (csl) Add csl.constants op and printing

### PR DESCRIPTION
**Pull request by AntonLydike**
Add the `csl.constants` op to define constant-initialized arrays and printing for said operation.

### CODE REVIEW

**Review by n-io — APPROVED** [review]
Looks good!

**Comment by AntonLydike** [comment]
@dk949 do you want to review this as well? Otherwise I'll just merge!

**Comment by dk949** [comment]
Sorry, completely missed it, can you give me 15min before merging?

File: tests/filecheck/backend/csl/print_csl.mlir [hunk 1]
```diff
@@ -389,6 +403,14 @@ csl.func @builtins() {
 // CHECK-NEXT:   return;
 // CHECK-NEXT: }
 // CHECK-NEXT:
+// CHECK-NEXT: fn constants() void {
+// CHECK-NEXT:   var v0 : [const27]i16 = @constants(const27, const27);
+// CHECK-NEXT:   const v1 : [const27]i16 = @constants(const27, const27);
+// CHECK-NEXT:   const v2 : [const27]i32 = @constants(const27, 100);
```
Comment (dk949) [inline tests/filecheck/backend/csl/print_csl.mlir:409]: Shouldn't this be `@constants([const27]i6, const27)`?

File: xdsl/backend/csl/print_csl.py [hunk 2]
```diff
@@ -476,6 +506,15 @@ def print_block(self, body: Block):
...
                         res, f"@concat_structs({a_var}, {b_var})"
                     )
+                case csl.ConstantsOp(
+                    size=size, value=val, result=res, is_const=constness
+                ):
+                    type = self._memref_type_to_string(res)
+                    res_name = self._get_variable_name_for(res)
+                    kind = "const" if constness else "var"
+                    self.print(
+                        f"{kind} {res_name} : {type} = @constants({self._var_use(size)}, {self._var_use(val)});"
```
Comment (dk949) [inline xdsl/backend/csl/print_csl.py:516]: I think the error for the above is here, the first arg should be an array type.

**Review by dk949 — CHANGES_REQUESTED** [review]
There's a small error in `@constants` printing, otherwise looks great :+1:

File: tests/filecheck/backend/csl/print_csl.mlir (hunk 1 above)
Comment (AntonLydike) [reply tests/filecheck/backend/csl/print_csl.mlir:409]: dayyyum, you are right! good catch!

**Review by dk949 — APPROVED** [review]

**Pull request merged**